fix(xtask): duplicated id generation

frenzox opened the pull request:
For some reason `update-ids` generates ids for gateway and gas-service twice on some branches even though the --no-dedupe option is not passed to the inner `cargo tree --workspace --depth 0`. This change ensures there are no duplicate crate names returned by the `workspace_crates_by_category` function.

roberts-ivanovs commented on offchain/components/relayer/src/includer.rs:
```diff
@@ -106,7 +106,8 @@ impl SolanaIncluder {
                     trace!("received a new message");
                     let approval = message.ok_or(IncluderError::ChannelClosed)?;
-                    futures.push(worker_future(approval));
+                    let worker_clone = worker.clone();
+                    futures.push(async move {worker_clone.include(approval).await});
```
this should work too
```suggestion
                    futures.push(worker.clone().include(approval));
```

frenzox committed:
fix(xtask): duplicated id generation

For some reason `update-ids` generates ids for gateway and gas-service
twice on some branches even though the --no-dedupe option is not passed.
This change ensures there are no duplicate crate names returned by the
`workspace_crates_by_category` function.

Signed-off-by: Guilherme Felipe da Silva <gfsilva.eng@gmail.com>

pierre-l commented on xtask/src/main.rs:
```diff
@@ -292,8 +292,8 @@ fn workspace_crates_by_category(
         .group_by(|(_, _, path)| path.contains("programs"));
-    let mut solana_programs = vec![];
-    let mut auxiliary_crates = vec![];
+    let mut solana_programs = HashMap::new();
+    let mut auxiliary_crates = HashMap::new();
```
I would use a `BTreeMap` here to avoid any ordering shenanigan.
Feel free to resolve this is if you feel it's irrelevant

frenzox replied:
yeah, we don't really care about the order here
